packages/server-core/src/media/static-resource/static-resource.hooks.ts

```diff
@@ -103,6 +103,39 @@ const createHashIfNeeded = async (context: HookContext<StaticResourceService>) =
   }
 }
 
+const updateName = async (context: HookContext<StaticResourceService>) => {
+  if (!context.data || !(context.method === 'create' || context.method === 'update' || context.method === 'patch')) {
+    throw new BadRequest(`${context.path} service only works for data in ${context.method}`)
+  }
+
+  if (Array.isArray(context.data)) throw new BadRequest('Batch create is not supported')
+
+  const id = context.id
+  const data = context.data
+  if (!data.key || !id) return
+
+  const [_, directory, file] = /(.*)\/([^\\\/]+$)/.exec(data.key)!
+  let doUpdate = false
```

Comment:
let's call this `shouldUpdateName` for clarity

Reply:
implemented